base/docs/basedocs.jl

```diff
@@ -1218,7 +1242,27 @@ getfield
     setfield!(value, name::Symbol, x)
 
 Assign `x` to a named field in `value` of composite type. The syntax `a.b = c` calls
-`setfield!(a, :b, c)`.
+`setfield!(a, :b, c)`. `value` must be mutable.
+
+# Examples
+```jldoctest
+julia> a = 1//2
```

Comment:
I would switch the examples to show a working example first. Also, perhaps make a custom `struct` too? It is not obvious that `Rational` has a `den` field.

Reply:
I was using `Rational` to make a connection with the `getfield` example directly above. I can switch to the field we want to access being `num` as in that example.